app/models/git_repository.rb

```diff
@@ -25,6 +25,29 @@ def self.delete_repo_dir(id, directory_name)
     FileUtils.rm_rf(directory_name)
   end
 
+  # Ping the git repository endpoint to verify that the network connection is still up.
+  # We use this approach for now over Rugged#check_connection because of a segfault:
+  #
+  #   https://github.com/libgit2/rugged/issues/859
+  #
+  def check_connection?
+    require 'net/ping/external'
+
+    # URI library cannot handle git urls, so just convert it to a standard url.
+    temp_url = url.dup
```

Comment:
Maybe a `url.dup.to_s` to avoid the `NoMethodError` that @d-m-u was reporting... though, how you get a `nil` `url` with a `GitRepository` is beyond me.

Reply:
Done